### ISSUE TITLE
re: Homebrew Cask Updater and Homebrew Updates

### ISSUE THREAD

**Issue by rkrug** [matryer/bitbar-plugins#493]
FAO: (@srdgh)
FAO: (@killercup)

I am using =these two plugins together, and it seems that when I am doing a Refresh All one of the two fails as both are trying to do a `brew update` which is not possible.

Would it be possible to add in one (or both) of these plugins a condition which waits for the `brew update` to finish and than continues?

Thanks,

Rainer

**Comment by iosdeveloper** [comment]
Use https://github.com/matryer/bitbar-plugins/blob/master/Dev/Homebrew/package_manager.7h.py?

**Comment by rkrug** [comment]
FAO: (@kdeldycke) @iosdeveloper 

I am trying it - but I get an error:

```
03:08:12 ~$ /Users/rainerkrug/BitBar/package_manager.7h.py
Traceback (most recent call last):
  File "/Users/rainerkrug/BitBar/package_manager.7h.py", line 534, in <module>
    print_menu()
  File "/Users/rainerkrug/BitBar/package_manager.7h.py", line 500, in print_menu
    map(methodcaller('sync'), managers)
  File "/Users/rainerkrug/BitBar/package_manager.7h.py", line 250, in sync
    if output.find('Not installed') == -1:
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 18: ordinal not in range(128)
```

Do I have to disable non-existing package managers?

**Comment by kdeldycke** [comment]
Looks like an issue about parsing unicode characters of Cask output.

Can you provides us with the result of the following commands:
```bash
$ python --version
$ brew cask list -1 | xargs -I "{}" brew cask info "{}"
```

**Comment by rkrug** [comment]
```
$ python --version
Python 2.7.12
```
The output from 
`$ brew cask list -1 | xargs -I "{}" brew cask info "{}"`
is at 
[https://gist.github.com/rkrug/b3cc88d7d92703213ff370a1ac614440](https://gist.github.com/rkrug/b3cc88d7d92703213ff370a1ac614440)

**Comment by kdeldycke** [comment]
Thanks @rkrug. Culprit likely to be: 
```
ubersicht: 1.0.42
Übersicht
http://tracesof.net/uebersicht
/usr/local/Caskroom/ubersicht/1.0.42 (0B)
https://github.com/caskroom/homebrew-cask/blob/master/Casks/ubersicht.rb
==> Contents
  Übersicht.app (app)
```
Source: https://gist.github.com/rkrug/b3cc88d7d92703213ff370a1ac614440#file-brew-cask-L625

**Comment by rkrug** [comment]
@kdeldycke I have just uninstalled Übersicht but had to zap it as it was still there. Now it is working. Nice. Is there a way to upgrade all? I only see it for ruby gems?

**Comment by tresni** [comment]
Only where the package manager actually has support for it.  Not all of them do.

— Brian

**Comment by rkrug** [comment]
OK. Thanks.

@kdeldycke Is there any chance that this bug can be fixed? It might also affect other packages?

**Comment by kdeldycke** [comment]
@rkrug : fixed by #516, waiting to be merged upstream.
